relay-config/src/config.rs

```diff
@@ -643,6 +700,160 @@ impl Config {
         Ok(config)
     }
 
+    /// Override configuration with values coming from other sources (e.g. env variables or
+    /// command line parameters)
+    pub fn override_config(
+        &mut self,
+        overrides: OverridableConfig,
+    ) -> Result<&mut Self, ConfigError> {
+        let relay = &mut self.values.relay;
+
+        if let Some(upstream) = overrides.upstream {
+            relay.upstream = UpstreamDescriptor::from_str(&upstream).map_err(|err| {
+                ConfigError::new_field_error(
+                    "upstream",
+                    upstream,
+                    err.context(ConfigErrorKind::InvalidOverride),
+                )
+            })?;
+        }
+
+        if let Some(host) = overrides.host {
+            relay.host = IpAddr::from_str(&host).map_err(|err| {
+                ConfigError::new_field_error(
+                    "host",
+                    host,
+                    err.context(ConfigErrorKind::InvalidOverride),
+                )
+            })?;
+        }
+
+        if let Some(port) = overrides.port {
+            relay.port = u16::from_str_radix(port.as_str(), 10).map_err(|err| {
+                ConfigError::new_field_error(
+                    "port",
+                    port,
+                    err.context(ConfigErrorKind::InvalidOverride),
+                )
+            })?;
+        }
+
+        let processing = &mut self.values.processing;
+        if let Some(enabled) = overrides.processing {
+            match enabled.to_lowercase().as_str() {
+                "true" => processing.enabled = true,
+                "false" => processing.enabled = false,
+                val => {
+                    return Err(ConfigError::new_field_error(
+                        "processing",
+                        val,
+                        Context::new(ConfigErrorKind::InvalidOverride),
+                    ));
+                }
+            }
+        }
+
+        // TODO are we going to support Cluster redis ?
```

Comment:
This is fine. It was explicitly defined that redis cluster is not supported for this type of configuration. Feel free to remove this TODO.